### PR TITLE
GRF-121

### PR DESCRIPTION
**Pull request by antoinetrouve**
**Description (for Contributor and Core Developer)**

Mock query get template (internal Api)

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ */
+class GetTemplateInMemory implements GetTemplate
+{
+    public function byUuid(string $templateId): ?Template
+    {
+        $templateUuid = TemplateUuid::fromString('02274dac-e99a-4e1d-8f9b-794d4c3ba330');
+
+        $template = new Template(
+            $templateUuid,
+            new TemplateCode('template_code'),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:35]:
```suggestion
            new TemplateCode('default_template'),
```

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            LabelCollection::fromArray(['fr_FR' => 'template_libelle']),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:36]:
```suggestion
            LabelCollection::fromArray(['en_US' => 'Default template']),
```

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                    AttributeIsLocalizable::fromBoolean(true),
+                    LabelCollection::fromArray(['fr_FR' => 'attribute_textarea_libelle']),
+                    $templateUuid
+                ),
+                AttributeRichText::create(
+                    AttributeUuid::fromString('840fcd1a-f66b-4f0c-9bbd-596629732950'),
+                    new AttributeCode('attribute_richtext_code'),
+                    AttributeOrder::fromInteger(3),
+                    AttributeIsLocalizable::fromBoolean(true),
+                    LabelCollection::fromArray(['fr_FR' => 'attribute_richtext_libelle']),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:60]:
```suggestion
                    LabelCollection::fromArray(['en_US' => 'Description']),
```

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                    new AttributeCode('attribute_textarea_code'),
+                    AttributeOrder::fromInteger(2),
+                    AttributeIsLocalizable::fromBoolean(true),
+                    LabelCollection::fromArray(['fr_FR' => 'attribute_textarea_libelle']),
+                    $templateUuid
+                ),
+                AttributeRichText::create(
+                    AttributeUuid::fromString('840fcd1a-f66b-4f0c-9bbd-596629732950'),
+                    new AttributeCode('attribute_richtext_code'),
+                    AttributeOrder::fromInteger(3),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:58]: The description should be the first attribute

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                    AttributeIsLocalizable::fromBoolean(true),
+                    LabelCollection::fromArray(['fr_FR' => 'attribute_richtext_libelle']),
+                    $templateUuid
+                ),
+                AttributeImage::create(
+                    AttributeUuid::fromString('8dda490c-0fd1-4485-bdc5-342929783d9a'),
+                    new AttributeCode('attribute_image_code'),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:65]:
```suggestion
                    new AttributeCode('banner_image'),
```

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                    AttributeOrder::fromInteger(4),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:66]: The banner image should be the second attribute

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+                    AttributeIsLocalizable::fromBoolean(false),
+                    LabelCollection::fromArray(['fr_FR' => 'attribute_image_libelle']),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:68]:
```suggestion
                    LabelCollection::fromArray(['en_US' => 'Banner image']),
```

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+        $template = new Template(
+            $templateUuid,
+            new TemplateCode('template_code'),
+            LabelCollection::fromArray(['fr_FR' => 'template_libelle']),
+            new CategoryId(1),
+            AttributeCollection::fromArray([
+                AttributeText::create(
+                    AttributeUuid::fromString('4873080d-32a3-42a7-ae5c-1be518e40f3d'),
+                    new AttributeCode('attribute_text_code'),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:41]:
```suggestion
                    new AttributeCode('seo_meta_title'),
```

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+                    AttributeOrder::fromInteger(1),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:42]: Seo Meta Title should be the 3rd attribute

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+                    AttributeIsLocalizable::fromBoolean(true),
+                    LabelCollection::fromArray(['fr_FR' => 'attribute_text_libelle']),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:44]:
```suggestion
                    LabelCollection::fromArray(['en_US' => 'SEO Meta Title']),
```

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+                    $templateUuid
+                ),
+                AttributeTextArea::create(
+                    AttributeUuid::fromString('69e251b3-b876-48b5-9c09-92f54bfb528d'),
+                    new AttributeCode('attribute_textarea_code'),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:49]:
```suggestion
                    new AttributeCode('seo_meta_description'),
```

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+                    AttributeOrder::fromInteger(2),
+                    AttributeIsLocalizable::fromBoolean(true),
+                    LabelCollection::fromArray(['fr_FR' => 'attribute_textarea_libelle']),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:52]:
```suggestion
                    LabelCollection::fromArray(['en_US' => 'SEO Meta Description']),
```

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+                    new AttributeCode('attribute_text_code'),
+                    AttributeOrder::fromInteger(1),
+                    AttributeIsLocalizable::fromBoolean(true),
+                    LabelCollection::fromArray(['fr_FR' => 'attribute_text_libelle']),
+                    $templateUuid
+                ),
+                AttributeTextArea::create(
+                    AttributeUuid::fromString('69e251b3-b876-48b5-9c09-92f54bfb528d'),
+                    new AttributeCode('attribute_textarea_code'),
+                    AttributeOrder::fromInteger(2),
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:50]: SEO Meta Description should be the 4th attribute

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+use Akeneo\Category\Domain\ValueObject\CategoryId;
+use Akeneo\Category\Domain\ValueObject\LabelCollection;
+use Akeneo\Category\Domain\ValueObject\Template\TemplateCode;
+use Akeneo\Category\Domain\ValueObject\Template\TemplateUuid;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class GetTemplateInMemory implements GetTemplate
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:27]: The **SEO Keywords** attribute is missing.
Could you add this attribute at the 5th position, please

Code: seo_keywords
Label: SEO Keywords
Type: textarea (simple)
Order: 5

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 1 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php]: Done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+                    AttributeUuid::fromString('69e251b3-b876-48b5-9c09-92f54bfb528d'),
+                    new AttributeCode('attribute_textarea_code'),
+                    AttributeOrder::fromInteger(2),
+                    AttributeIsLocalizable::fromBoolean(true),
+                    LabelCollection::fromArray(['fr_FR' => 'attribute_textarea_libelle']),
+                    $templateUuid
+                ),
+                AttributeRichText::create(
+                    AttributeUuid::fromString('840fcd1a-f66b-4f0c-9bbd-596629732950'),
+                    new AttributeCode('attribute_richtext_code'),
```
[the remark this answers is not in the document]
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php]: Done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 4 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:58]: Done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 5 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php]: Done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 6 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:66]: Done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 7 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php]: Done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 11 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php]: Done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 12 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php]: Done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 14 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:27]: Done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 10 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php]: Done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 13 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php:50]: Done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 2 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php]: Done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 3 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php]: Done

File: src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/routing.yml [hunk 16]
```diff
@@ -9,3 +9,6 @@ internal_api_category_legacy:
 internal_api_category:
     resource: '@AkeneoCategoryBundle/Resources/config/routing/internal_api/category.yml'
     prefix: /category
+
+internal_api_template:
+    resource: '@AkeneoCategoryBundle/Resources/config/routing/internal_api/category_template.yml'
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/routing.yml:14]: Missing prefix parameter. It should fix the errors on CI related to the not found internal API endpoint for Rules domain

File: src/Akeneo/Category/back/tests/Integration/Infrastructure/Storage/InMemory/GetTemplateInMemoryIntegration.php [hunk 17]
```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Category\back\tests\Integration\Infrastructure\Storage\InMemory;
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/tests/Integration/Infrastructure/Storage/InMemory/GetTemplateInMemoryIntegration.php:5]:
```suggestion
namespace Akeneo\Test\Category\Integration\Infrastructure\Storage\InMemory;
```

File: src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/routing.yml [hunk 18]
```diff
@@ -9,3 +9,6 @@ internal_api_category_legacy:
 internal_api_category:
     resource: '@AkeneoCategoryBundle/Resources/config/routing/internal_api/category.yml'
     prefix: /category
+
+internal_api_template:
```
Comment (olivierpontier) [inline src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/routing.yml:13]:
```suggestion
internal_api_category_template:
```

**Review by BryanS29-akeneo — APPROVED** [review]

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 2 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php]: done

File: src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php (hunk 3 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/InMemory/GetTemplateInMemory.php]: Done

File: src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/routing.yml (hunk 16 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/routing.yml:14]: prefix : /category/template added

Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/routing.yml:14]: done

File: src/Akeneo/Category/back/tests/Integration/Infrastructure/Storage/InMemory/GetTemplateInMemoryIntegration.php (hunk 17 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/tests/Integration/Infrastructure/Storage/InMemory/GetTemplateInMemoryIntegration.php]: done

File: src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/routing.yml (hunk 18 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Symfony/Resources/config/routing.yml]: done

**Review by olivierpontier — APPROVED** [review]

**Pull request merged**